datasources/src/main/elm/sources/Update.elm

```diff
@@ -58,16 +59,35 @@ update msg model =
           (model,Cmd.none)
 
     SaveCall d ->
-      case model.mode of
-        ShowDatasource _ Nothing ->
-          let
-            dataSource = {d | id = if String.isEmpty d.id then d.name else d.id }
-          in
-            (model, saveDataSource model True dataSource)
-        ShowDatasource _ (Just _) ->
-          (model, saveDataSource model False d)
-        _ ->
-          (model, Cmd.none)
+      let
+         listDatasources = model.dataSources
+      in
+        case model.mode of
+          -- Creation scenario
+          ShowDatasource _ Nothing ->
+            let
+              dataSource = {d | id = if String.isEmpty d.id then d.name else d.id }
+            in
+              ( { model | dataSources = dataSource :: listDatasources }
```

Comment:
When we create a Datasource, the left menu was not updated withe newly created datasource